certora/specs/SocialRecoveryModule.spec

```diff
@@ -1,26 +1,215 @@
+    assert guardianStorageContract.entries[safeContract].guardians[prevGuardian] == nextGuardian;
+    assert guardian != otherAccount => otherAccountIsGuardian == currentContract.isGuardian(safeContract, otherAccount);
+    assert currentGuardiansCount > 0 => currentGuardiansCount - 1 == to_mathint(guardianStorageContract.entries[safeContract].count);
+    assert threshold <= guardianStorageContract.entries[safeContract].count;
+}
+
+// This integrity rule verifies that the guardian can always be revoked considering ideal conditions.
+rule guardianCanAlwaysBeRevoked(env e, address guardian, address prevGuardian, uint256 threshold) {
     requireSocialRecoveryModuleEnabled();
+    requireGuardiansLinkedListIntegrity(guardian);
+
+    // No value should be sent with the transaction.
+    require e.msg.value == 0;
+    // If new threshold is 0, then you must be removing the last guardian meaning the guardian count should be 1.
+    require threshold == 0 => guardianStorageContract.entries[safeContract].count == 1;
+    // The new threshold should be less than or equal to the guardian count after removing.
+    require to_mathint(threshold) <= guardianStorageContract.entries[safeContract].count - 1;
+    // The address should be a guardian.
+    require currentContract.isGuardian(safeContract, guardian);
+
+    address nextGuardian = guardianStorageContract.entries[safeContract].guardians[guardian];
+    require guardianStorageContract.entries[safeContract].guardians[prevGuardian] == guardian;
+
+    uint256 currentGuardiansCount = guardianStorageContract.entries[safeContract].count;    
+
+    // Safe Contract should be the sender of the transaction.
+    require e.msg.sender == safeContract;
+    currentContract.revokeGuardianWithThreshold@withrevert(e, safeContract, prevGuardian, guardian, threshold);
+    bool isReverted = lastReverted;
+
+    assert !isReverted &&
+        guardianStorageContract.entries[safeContract].guardians[prevGuardian] == nextGuardian &&
+        !currentContract.isGuardian(safeContract, guardian) &&
+        currentGuardiansCount - 1 == to_mathint(guardianStorageContract.entries[safeContract].count);
+}
+
+// This integrity rule verifies the possibilites in which the revocation of a new guardian can revert.
+rule revokeGuardianRevertPossibilities(env e, address prevGuardian, address guardian, uint256 threshold) {
+    requireGuardiansLinkedListIntegrity(guardian);
+
+    bool isGuardian = currentContract.isGuardian(safeContract, guardian);
 
-    safeContract.disableModule@withrevert(e, prevModule, currentContract);
+    currentContract.revokeGuardianWithThreshold@withrevert(e, safeContract, prevGuardian, guardian, threshold);
     bool isReverted = lastReverted;
 
-    assert !isReverted => !safeContract.isModuleEnabled(currentContract);
+    assert isReverted =>
+        !isGuardian ||
+        e.msg.sender != safeContract ||
+        e.msg.value != 0 ||
+        guardian == 0 ||
+        guardian == SENTINEL() ||
```

Comment:
```suggestion
```

I think this should also be removed - `isGuardian(guardian)` implies that it isn't address 0 or sentinel right?

Reply:
Correct, updated 👍🏾